File: rest-api/src/main/java/com/kt/kol/api/comment/model/Comment.java
package com.kt.kol.api.comment.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("comments")
public class Comment {
    @Id
    private Long id;

    private String content;
    private Long postId;
    private Long userId;
    private Long parentId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: rest-api/src/main/kotlin/com/kt/kol/api/comment/model/Comment.kt
package com.kt.kol.api.comment.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table("comments")
data class Comment(
    @Id
    val id: Long? = null,
    var content: String? = null,
    var postId: Long? = null,
    var userId: Long? = null,
    var parentId: Long? = null,
    var createdAt: LocalDateTime? = null,
    var updatedAt: LocalDateTime? = null
)